### PR TITLE
[IMP] mail: improve o_MessageActionList design

### PR DESCRIPTION
**Pull request by Nemael**
Improves the design of o_MessageActionList

Task-2793394

### CODE REVIEW

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Would keep exact same style in mobile.

In desktop, just shrink menu.

File: addons/mail/static/src/components/message_action_list/message_action_list.xml [hunk 1]
```diff
@@ -3,17 +3,17 @@
     <t t-name="mail.MessageActionList" owl="1">
         <t t-if="messageActionList">
             <div class="o_MessageActionList d-flex border rounded-sm bg-white" t-attf-class="{{ className }}" t-on-click="messageActionList.onClick" t-ref="root">
-                <i t-if="messageActionList.message.hasReactionIcon" class="o_MessageActionList_action o_MessageActionList_actionReaction fa fa-lg fa-smile-o p-2 o_cursor_pointer" t-att-title="messageActionList.addReactionText" t-on-click="messageActionList.onClickActionReaction" t-ref="actionReaction"/>
+                <i t-if="messageActionList.message.hasReactionIcon" class="o_MessageActionList_action o_MessageActionList_actionReaction fa fa-lg fa-smile-o o_cursor_pointer" t-attf-class="{{messaging.device.isSmall ? 'p-2' : 'p-1'}}" t-att-title="messageActionList.addReactionText" t-on-click="messageActionList.onClickActionReaction" t-ref="actionReaction"/>
```
Comment (alexkuhn) [inline addons/mail/static/src/components/message_action_list/message_action_list.xml:6]: for styling, we prefer inner-whitespace with `{{ }}`, e.g. `{{  'p-2' }}` instead of `{{'p-2'}}`

File: addons/mail/static/src/components/message_action_list/message_action_list.scss [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
 }
 
 .o_MessageActionList_action:hover {
-    background-color: mix($border-color, $white);
+  background-color: mix($border-color, $white);
 }
 
 .o_MessageActionList_actionStar.o_MessageActionList_actionStar_active {
     color: $o-main-favorite-color;
+    opacity: 1;
```
Comment (alexkuhn) [inline addons/mail/static/src/components/message_action_list/message_action_list.scss]: Would remove `opacity` altogether.

File: addons/mail/static/src/components/message_action_list/message_action_list.scss [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+}
+
 .o_MessageActionList:hover {
-    box-shadow: 0 4px .5rem -.5rem $black;
+  box-shadow: 0 4px .5rem -.5rem $black;
+}
+
+.o_MessageActionList_action {
+  opacity: 0.4;
+  padding: map-get($spacers, 1) map-get($spacers, 2);
```
Comment (alexkuhn) [inline addons/mail/static/src/components/message_action_list/message_action_list.scss]: I think you can simply remove these `padding` rules, as inline bootstrap class `p-2` and `p-1` has precedence

**Comment by alexkuhn** [comment]
Don't forget to rebase to solve conflict 😉

File: addons/mail/static/src/components/message_action_list/message_action_list.scss [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
+  box-shadow: 0 4px .5rem -.5rem $black;
+}
+
+.o_MessageActionList_action {
+  padding: map-get($spacers, 1) map-get($spacers, 2);
 }
 
 .o_MessageActionList_action:hover {
-    background-color: mix($border-color, $white);
+  background-color: mix($border-color, $white);
```
Comment (alexkuhn) [inline addons/mail/static/src/components/message_action_list/message_action_list.scss:15]: unnecessary diff. Indent is 4 whitespaces overall

File: addons/mail/static/src/components/message_action_list/message_action_list.scss (hunk 3 above)
Comment (Nemael) [reply addons/mail/static/src/components/message_action_list/message_action_list.scss]: These css padding rules acutally still have an effect on the padding, and I couldn't reproduce that effect with added `p-3` and `p-1` rules.
So I left that first css padding rule in :)

Comment (alexkuhn) [reply addons/mail/static/src/components/message_action_list/message_action_list.scss]: Indeed, the 1st css had no classname equivalent.
But you can make it with bootstrap classnames, here it's equivalent to `py-1 px-2` I think.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
